diff --git a/src/quick/scenegraph/qsgdistancefieldglyphnode_p.cpp b/src/quick/scenegraph/qsgdistancefieldglyphnode_p.cpp
index 3852b01..6301281 100644
--- a/src/quick/scenegraph/qsgdistancefieldglyphnode_p.cpp
+++ b/src/quick/scenegraph/qsgdistancefieldglyphnode_p.cpp
@@ -93,9 +93,9 @@ const char *QSGDistanceFieldTextMaterialShader::fragmentShader() const {
         "uniform highp float alphaMin;                                               \n"
         "uniform highp float alphaMax;                                               \n"
         "void main() {                                                               \n"
+        "    mediump float a = texture2D(texture, sampleCoord).a;                    \n"
         "    gl_FragColor = color * smoothstep(alphaMin,                             \n"
-        "                                      alphaMax,                             \n"
-        "                                      texture2D(texture, sampleCoord).a);   \n"
+        "                                      alphaMax, a);                         \n"
         "}";
 }
 
@@ -473,10 +473,10 @@ const char *DistanceFieldShiftedStyleTextMaterialShader::fragmentShader() const
             "uniform highp float alphaMin;                                                         \n"
             "uniform highp float alphaMax;                                                         \n"
             "void main() {                                                                         \n"
-            "    highp float a = smoothstep(alphaMin, alphaMax, texture2D(texture, sampleCoord).a);\n"
-            "    highp vec4 shifted = styleColor * smoothstep(alphaMin,                            \n"
-            "                                                 alphaMax,                            \n"
-            "                                                 texture2D(texture, shiftedSampleCoord).a); \n"
+            "    mediump float sa = texture2D(texture, sampleCoord).a;                             \n"
+            "    highp float a = smoothstep(alphaMin, alphaMax, sa);                               \n"
+            "    mediump float ssa = texture2D(texture, shiftedSampleCoord).a;                     \n"
+            "    highp vec4 shifted = styleColor * smoothstep(alphaMin, alphaMax, ssa);            \n"
             "    gl_FragColor = mix(shifted, color, a);                                            \n"
             "}";
 }
